### DISCUSSION TITLE
2022-02-28 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1102] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254586)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254587)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254589)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254590)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254591)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254592)
- [Google Summer of Code prep](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2254594)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2265368)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2266122)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2266903)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2267608)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2268509)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2268859)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1102#discussioncomment-2272037)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- **Specs**:
  - [DRAFT] https://github.com/centerofci/mathesar-wiki/pull/36

### Complete
- **Milestones**:
  - [2022-02 improvements](https://github.com/centerofci/mathesar/milestone/43) with 8 issues complete
- **Discussions**:
  - https://github.com/centerofci/mathesar/discussions/1081
- **Specs**:
  - https://github.com/centerofci/mathesar-wiki/pull/35
- **Wiki updates**:
  - The Mathesar types engineering page was renamed to [UI Types](https://wiki.mathesar.org/en/engineering/architecture/ui-types) (and the text was updated throughout to refer to "UI types") to keep the nomenclature consistent with our code.
  - The [Product Spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views) has had major updates, but isn't complete yet.
- **Full stack and general work**:
  - https://github.com/centerofci/mathesar/pull/1062
- **Backend work**:
  - https://github.com/centerofci/mathesar/pull/1088
  - https://github.com/centerofci/mathesar/pull/1091

**Comment by kgodey** [comment]
## Frontend priorities

1. Complete front end changes to https://github.com/centerofci/mathesar/pull/1047
2. Complete #775
3. Get open draft PRs complete and merged:
    1. #1076
    2. #1080
4. Make progress on issues from the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) and [2022-03 improvements](https://github.com/centerofci/mathesar/milestone/44) milestones. Some ideas: 
    1. #970
    2. #927
    3. #928
    4. #929
    5. #936
    6. #659
    7. #257 (assigned to @pavish)

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good. With the difficulty I've had thus far in predicting the time-to-completion of my refactoring project, I'll refrain from continuing to make such predictions. But I'll say that I think I'm really _really_ close to completing it!

**Reply by pavish** [reply]
Looks good to me.

I would like to schedule improvements with regards to editing the table after we have basic data type frontend implementation completed. Each type has different interactions and would possibly require some refactoring, building these improvements over existing code might result in more code to refactor. This includes some suggestions mentioned above:
* #927  
* #928 
* #929 
* #936

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP.

We divided up the issues as follows:

1. @dmos62:
    1. **Filtering**:
        1. **Get open PRs merged**:
            1. https://github.com/centerofci/mathesar/pull/1079
            2. https://github.com/centerofci/mathesar/pull/1087
            3. https://github.com/centerofci/mathesar/pull/1089
            4. https://github.com/centerofci/mathesar/pull/1090
            5. https://github.com/centerofci/mathesar/pull/1098
            6. https://github.com/centerofci/mathesar/pull/1101
        2. **Other filtering issues**:
            1. https://github.com/centerofci/mathesar/issues/1100
            2. #1097 
    1. **Data Type API updates:**:
        1. **Get open PRs merged**:
            1. https://github.com/centerofci/mathesar/pull/1093
        3. **Work on issues**:
            1. https://github.com/centerofci/mathesar/issues/1036
            2. https://github.com/centerofci/mathesar/issues/1037
2. @mathemancer:
    1. Final updates to https://github.com/centerofci/mathesar/issues/845
    2. https://github.com/centerofci/mathesar/issues/1094
    4. **Grouping**:
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        5. https://github.com/centerofci/mathesar/issues/384
        6. https://github.com/centerofci/mathesar/issues/429
        7. https://github.com/centerofci/mathesar/issues/419
        8. https://github.com/centerofci/mathesar/issues/432 (currently blocked)
3. @silentninja:
    1. **Replace column name usage with ID:**
        1. Get #1047 merged
        2. https://github.com/centerofci/mathesar/issues/839
            1. https://github.com/centerofci/mathesar/issues/992
                1. PR: https://github.com/centerofci/mathesar/pull/1104
            3. https://github.com/centerofci/mathesar/issues/993
            4. https://github.com/centerofci/mathesar/issues/994
            5. https://github.com/centerofci/mathesar/issues/995
        3. #996   
    2. https://github.com/centerofci/mathesar/issues/1064
        1. PR: https://github.com/centerofci/mathesar/pull/1103 
    3. https://github.com/centerofci/mathesar/issues/1105 (currently blocked)

I'll also work on #1008 if I have time this week.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Comment by kgodey** [comment]
## Design priorities

1. #1063 is top priority since it's blocking backend work.
2. Exploratory wireframing for #1065
3. Design discussions for #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
@kgodey Looks good. I have already coordinated the weekly check-in's w/ @ppii775

**Comment by kgodey** [comment]
## Product priorities

1. Finalize [product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views).
2. Updates to the GitHub issues for Views design to conform to the spec.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-02-28 to 2022-03-04 | Brent |
| 2022-03-07 to 2022-03-11 | Kriti |
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Dominykas |
| 2022-04-11 to 2022-04-15 | Ghislaine |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by kgodey** [comment]
## Google Summer of Code prep

Google Summer of Code organizations are going to be announced on Monday, Mar 7 (in one week). If we do get selected, we will have an influx of people looking for issues to contribute to. We need a bunch of `good first issue` and `help wanted` issues. Also, the Comms Assignee should expect to be spending more time on comms.

It would be great if everyone could take an hour this week to look through our issues and:
- make sure the existing `good first issue` and `help wanted` issues look good.
- create any new  `good first issue` and `help wanted` issues that you can think of.

**Comment by dmos62** [comment]
> How did your most recent work go?

Nothing out of the ordinary. Progress seemed smooth.

>What are you planning to work on soon?

See filtering PRs merged. Work out new tasks that stem from discussions of those PRs (I've been trying to make large requested changes in new PRs).

Go on to data type API updates after that.

>Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Not so great. In preparation for moving the front end to use column ids, I got sucked into a refactor that has ended up being way more involved than I predicted. Once we get that work merged and the dust settles a bit, I think it will be good for @pavish and I (and perhaps @kgodey too) do do some retrospective analysis on our process. I'm hoping this refactoring will be worth it in the end, but it doesn't feel in-line with our current goals of prioritizing features.
- The good news is that some of the other recent front-end architectural improvements (i.e. TS strict, Prettier, E2E testing) has made this refactoring work quicker/easier than it otherwise would have been.
- There are still a couple nagging architectural issues that have been slowing me down though, and I plan to discuss them with @pavish on a sync call at some point (though it's lower-priority).

> What are you planning to work on soon?

- Finish my refactor. To aid review, I'm planning to open a PR against `master` with this work (instead of tacking it all in #1047 which originally motivated this work).
- Add front end changes #1047. I want to base those changes on the work already in #1047 _and_ the refactoring work that I'd like to PR against `master`. I'm open to suggestions for the best workflow. Unless I hear otherwise, I'll probably do my work locally on top of cherry-picked commits, then cherry pick my work into a public branch once my refactoring work is merged into `master` (and thus can be cleanly merged into #1047).

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope.

**Reply by kgodey** [reply]
> I think it will be good for @pavish and I (and perhaps @kgodey too) do do some retrospective analysis on our process.

I'm happy to be involved in this if you think it would be be helpful.

> I'm open to suggestions for the best workflow. 

Why not open the refactoring PR, work on something else until it is merged, and then merge `master` into #1047 and go from there? If you'd like to use your refactoring work in the "something else" you're working on, you can base that on top of the refactoring branch since you won't need backend work from #1047 in it.

**Reply by seancolsen** [reply]
@kgodey 

> work on something else until it is merged

Cool, will do.

**Comment by kgodey** [comment]
## Kriti's check in

> How did your most recent work go?

I did get a significant amount of work done on specifying formulas for the [Views product spec](https://wiki.mathesar.org/en/product/specs/2022-01-views), but I'm not done yet. I spent too much time in meetings last week (>10 hours in a 4 day week) to be able to get much focused work done (especially in addition to reviews, discussion, etc.) I'm rearranging my schedule so I have fewer meetings each week.

> What are you planning to work on soon?

My list is the same as last week with the additional task of looking through our contributor-friendly issues:
- Complete work on the Views product spec
- Spend some time working on `good first issue` and `help wanted` issues
- Open a new discussion for Views product spec review
- Update design tickets for Views based on product spec
- #1008 if time permits

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.

**Comment by mathemancer** [comment]
## Brent's check in

> How did your most recent work go?

The week was productive, but initial types is starting to feel like Zeno's Milestone.  We're down to fiddly details there, and every time something is solved, we seem to discover more that needs to be done.

> What are you planning to work on soon?

Once I get the custom money PR submitted, I need to tidy up and add a few details to the date/time spec while I still remember them.  Then on to grouping.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nothing at the moment.

**Reply by kgodey** [reply]
> initial types is starting to feel like Zeno's Milestone. 

It's always been the biggest milestone in the roadmap, there's so much to do for each type. I expect a lot of issues to be closed this week, there's a number of pull requests in review. I expect that _someone_ on the backend team can move on to the next milestone next week.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

It went well. Regular check-ins with Kriti are helpful to unblock design work. Money-type related discussions during the team meeting were also helpful in resolving the spec updates.

> What are you planning to work on soon?

#1065 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

Pretty well! Text type implementation is ready, along with groundwork for possible cell types with varying interactions. The same PR also contains basic boolean type implementation without display_options. 

> What are you planning to work on soon?

* Complete boolean type support
* Number component improvements
* Number type frontend implementation

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None at the moment

**Comment by silentninja** [comment]
> How did your most recent work go?

It was quite slow, as I had a runny nose and couldn't concentrate much. Though it is taking longer than I expected, the issues are nearing completion, so not much to fret about. 

> What are you planning to work on soon?

- Finish issue #839 and related sub-issues
- #1105 
- Start working on issues in Linking tables milestone

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None
